frontend/scripts/containers/shared/search-react.container.js

```diff
@@ -0,0 +1,49 @@
+import { bindActionCreators } from 'redux';
+import { connect } from 'preact-redux';
+import groupBy from 'lodash/groupBy';
+import Search from 'react-components/shared/search.component.js';
+import { selectExpandedNode, selectNode } from 'actions/tool.actions';
+import { ACTORS_COLUMN_IDS, IMPORTER_EXPORTER_TYPE } from 'constants';
+
+let searchNodes;
+
+const mapStateToProps = (state) => {
+  const { nodes, selectedNodesIds } = state.tool;
+  // store nodes at container level to avoid rerendering when filtering... for want of a better solution
+  if (nodes !== undefined && (!searchNodes || nodes.length !== searchNodes.length)) {
+    const allNodes = nodes.filter(
+      node => node.hasFlows === true &&
+      node.isAggregated !== true &&
+      node.isUnknown !== true
+    );
+
+    searchNodes = Object.values(groupBy(allNodes.filter(x => ACTORS_COLUMN_IDS.includes(x.columnId)), 'mainNodeId'))
```

Comment:
allNodes already specifies which type of profile each node has, so you can use that instead of the 3xEvil constant